source/common/http/header_map_impl.cc

```diff
@@ -525,6 +507,18 @@ size_t HeaderMapImpl::removePrefix(const LowerCaseString& prefix) {
   return old_size - headers_.size();
 }
 
+size_t HeaderMapImpl::remove(const LowerCaseString& key) {
+  return HeaderMapImpl::removeIf([&key](const HeaderEntry& entry) -> bool {
+    return key.get() == entry.key().getStringView();
+  });
+}
+
+size_t HeaderMapImpl::removePrefix(const LowerCaseString& prefix) {
+  return HeaderMapImpl::removeIf([&prefix](const HeaderEntry& entry) -> bool {
+    return absl::StartsWith(entry.key().getStringView(), prefix.get());
```

Comment:
just a quick note that we can make both of these functions fast if we use std::map as a lazy associative structure (as opposed to a hash-map).